### PR TITLE
CRIMRE-210 user invitations expire in 48 hours

### PR DESCRIPTION
**Pull request by timpeat**
## Description of change

User account invitations only last 48 hours from creation.

## Link to relevant ticket
[CRIMRE-210](https://dsdmoj.atlassian.net/browse/CRIMRE-210)

## Notes for reviewer

Note this PR only adds a time limit to invitations and prevents authentication if more than 48 hours have expired since the invitation was created.
 
How a user manager deals with expired invitations will be part of another ticket. As will the text for the invitation email.

## How to manually test the feature

[CRIMRE-210]: https://dsdmoj.atlassian.net/browse/CRIMRE-210?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,11 +1,24 @@
 class User < ApplicationRecord
   devise :omniauthable, :timeoutable
+
   include Reauthable
 
+  before_create do
+    self.invitation_expires_at = Rails.configuration.x.auth.reauthenticate_in.from_now
+  end
+
+  scope :pending_activation, lambda {
```
Comment (willmcb) [inline app/models/user.rb:10]: Wondering if it would be good to add a spec for the scopes here.

**Review by willmcb — APPROVED** [review]
LGTM - see comment around adding a spec for the sopes 👍

File: app/models/user.rb [hunk 2]
```diff
@@ -1,11 +1,24 @@
 class User < ApplicationRecord
   devise :omniauthable, :timeoutable
+
   include Reauthable
 
+  before_create do
+    self.invitation_expires_at = Rails.configuration.x.auth.reauthenticate_in.from_now
```
Comment (zheileman) [inline app/models/user.rb:7]: Should this use the `auth.invitation_ttl` config instead?

Comment (timpeat) [reply app/models/user.rb:7]: @zheileman great catch! Thank you!

File: app/models/user.rb (hunk 1 above)
Comment (timpeat) [reply app/models/user.rb:10]: done

**Pull request merged**